v2/src/datadog.ts

```diff
@@ -201,6 +200,19 @@ function grantReadLambdasFromSecretArn(construct: Construct, arn: string, lambda
   });
 }
 
+function extractSingletonFunctions(lambdaFunctions: LambdaFunction[]) {
+  // extract lambdaFunction property from Singleton Function
+  // using any here since lambdaFunction is a private property
```

Comment:
I think you can get around the any cast by using the bracket notation to access the lambdaFunction property https://dev.to/zirkelc/hack-access-private-class-members-in-typescript-41jp

Reply:
It looks like this `dot-notation` rule treats the bracket notation as an error.

https://github.com/DataDog/datadog-cdk-constructs/blob/0bf6d168ed5aaa6615a78021fc2aa5aa4cf60c32/v2/.eslintrc.json#L197-L199

But based on the article you shared I made a minor update to at least limit the any cast to only Singleton functions instead of all functions. Let me know if that's acceptable

Reply:
Ah I see, that's trickier than I thought. Since the typechecking can help us avoid runtime errors and the linting is more just for a consistent style, I would do a single-line eslint override here to allow the bracket notation

Reply:
Got it, I changed the type cast to a lambda.SingletonFunction and added the single-line eslint override for the dot-notation rule

The cast is still needed because the lambdaFunction property is only on the lambda.SingletonFunction type and not on the lambda.Function, lambdaNodejs.NodejsFunction, or lambdaPython.PythonFunction types (which are all part of the custom LambdaFunction type).

<img width="851" alt="Screenshot 2023-09-19 at 5 15 13 PM" src="https://github.com/DataDog/datadog-cdk-constructs/assets/35278470/e03b81f9-67ea-4f3e-9aac-f8a7a734df58">

Reply:
Refactored to use the type guard as we discussed